src/features/users/components/users-overview.tsx

```diff
@@ -0,0 +1,206 @@
+import { Box, Divider, Grid, IconButton, MenuItem, Paper, TextField, Typography } from "@mui/material";
+import { useEffect, useState } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import { User } from "../../../app/models/User";
+import { RootState } from "../../../app/store";
+import { addUser } from "../users-thunk";
+import { UserListItem } from "./user-list-item";
+import { useFormik } from "formik";
+import FormikInputField from "../../../common-components/formik-field";
+import { STATUSES, GENDERS } from "../../../constants";
+import { validationSchema } from "../../../app/commonSchemas.ts/schemas";
+import AddIcon from '@mui/icons-material/Add';
+import CloseIcon from '@mui/icons-material/Close';
+import CheckIcon from '@mui/icons-material/Check';
+import axios, { AxiosResponse } from "axios";
+import { config } from "../../../axiosHelpers/axiosPutHelper";
+import { setUsers } from "../users-slice";
+
+
+export const UsersOverview = () => {
+  const [isAdd, setIsAdd] = useState(false);
+  const [isSaveButtonEnabled, setIsSaveButtonEnabled] = useState(false);
+  const [localUsersData, setLocalUsersData] = useState([])
+  const usersList = useSelector((state: RootState) => state.users.usersList);
+
+  const dispatch = useDispatch();
+
+  const getUsers = async (): Promise<AxiosResponse> => {
+    let result = {} as AxiosResponse;
+
+    return await axios.get('https://gorest.co.in/public/v2/users', config)
+    .then(function (response) {
+      result.data = response.data;
+      setLocalUsersData(response.data);
+      return result;
+    });
+  }
+
+  useEffect(() => {
+    if (localUsersData?.length === 0 || usersList?.length === 0) {
+      getUsers();
+    }
+
+  }, [localUsersData, usersList])
+
+  useEffect(() => {
+
+    if (localUsersData?.length > 0) {
+      dispatch(setUsers(localUsersData));
+    }
+    }, [dispatch, localUsersData])
+
+  const initialValuesAdd = {
+    email: '',
+    name: '',
+    status: '',
+    gender: 'male',
+  };
+
+  const checkIfIsValid = (value: any) =>
+    validationSchema.validate(value)
+    .then(() => setIsSaveButtonEnabled(true))
+    .catch(() => setIsSaveButtonEnabled(false));
+
+  const formik = useFormik({
+    initialValues: initialValuesAdd,
+    validationSchema,
+    validate: checkIfIsValid,
+    onSubmit: () => {},
+  });
+
+  const handleSaveNewUser = () => {
+    const postBody = {
+      name: formik.values?.name,
+      email: formik.values?.email,
+      gender: formik.values?.gender,
+      status: formik.values?.status,
+    } as User
+
+    dispatch(addUser(postBody))
+    setIsAdd(false);
+  }
+
+  const addNewUserView = (
+    <div>
+      <Box padding={2} paddingBottom={4}>
+        <Paper elevation={4}>
+          <Grid container spacing={2} paddingLeft={1} paddingRight={1}>
```

Comment:
Add additional padding to the bottom of this container to give the crud inputs more space.

`<Grid container spacing={2} padding={1} paddingBottom={4}>`